File: Advanced_Prog_Assignment3-csharp/EmployeeManagementSystem/Exceptions/InvalidEmployeeIdException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmployeeManagementSystemCSharp.Exceptions
{
    public class InvalidEmployeeIdException : Exception
    {
        public InvalidEmployeeIdException(string message)
            : base(message)
        {
        }

        public static void validateEmployeeID(string employeeID)
        {
            try
            {
                if (employeeID == null)
                {
                    throw new InvalidEmployeeIdException("Employee ID cannot be null.");
                }

                if (employeeID.Length < 5)
                {
                    throw new InvalidEmployeeIdException("Employee ID must be at least 5 characters long.");
                }

                Console.WriteLine("✓ Employee ID is valid: " + employeeID);
            }
            catch (InvalidEmployeeIdException e)
            {
                Console.Error.WriteLine("Error: " + e.Message);
            }
        }
    }
}

File: ASSIGNMENT3/EmployeeManagementSystem/src/exceptions/InvalidEmployeeIdException.java
package exceptions;

public class InvalidEmployeeIdException extends Exception {
    
    public InvalidEmployeeIdException(String message) {
        super(message);
    }

    public static void validateEmployeeID(String employeeID) {
        try {
            if (employeeID == null) {
                throw new InvalidEmployeeIdException("Employee ID cannot be null.");
            }
            if (employeeID.length() < 5) {
                throw new InvalidEmployeeIdException("Employee ID must be at least 5 characters long.");
            }
            System.out.println("✓ Employee ID is valid: " + employeeID);
            
        } catch (InvalidEmployeeIdException e) {
            System.err.println("❌ Error: " + e.getMessage());
        }
    }
}